src/main/java/fi/aalto/cs/apluscourses/model/APlusAuthentication.java

```diff
@@ -1,27 +1,57 @@
 package fi.aalto.cs.apluscourses.model;
 
 import java.util.Arrays;
-
 import org.apache.http.HttpRequest;
 import org.jetbrains.annotations.NotNull;
 
-public class APlusAuthentication {
+public class APlusAuthentication implements Authentication {
   @NotNull
   private final char[] token;
+  private final int maxLength;
+  private int length = 0;
 
-  /**
-   * Initializes an instance with the given token. Note, that the given token array is not cleared
-   * or overwritten.
-   */
-  public APlusAuthentication(@NotNull char[] token) {
-    this.token = token.clone();
+  public APlusAuthentication(int maxLength) {
+    this.token = new char[maxLength];
+    this.maxLength = maxLength;
```

Comment:
I'm not sure if this single-array-instance limitation actually brings any benefit in this case (while imposing an arbitrary length limit)? Then again if the tokens are always < 40 it's no big deal either way

Reply:
If the concern is that the old array remains somewhere, then it could always be cleared when a new array is allocated

Reply:
Now when I thought about it, I see you were right. I reverted to use "create new authentication object for new token" implementation we had. It's certainly better.

Reply:
this is to change for being stored in the IJ's `PasswordSafe` in a way, that there is not gonna be a token `char []` at all